sdk/tests/deqp/functional/gles3/es3fDefaultVertexAttributeTests.js

```diff
@@ -0,0 +1,541 @@
+        return [value[0], value[1], 0, 1];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttrib3fv = function() {
+    this.caseName = 'vertex_attrib_3fv';
+    this.name = 'VertexAttrib3fv';
+    this.signed = true;
+    this.load = function(index, value) {
+        gl.vertexAttrib3fv(index, value.slice(0, 3));
+        return [value[0], value[1], value[2], 1];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttrib4fv = function() {
+    this.caseName = 'vertex_attrib_4fv';
+    this.name = 'VertexAttrib4fv';
+    this.signed = true;
+    this.load = function(index, value) {
+        gl.vertexAttrib4fv(index, value.slice(0, 4));
+        return [value[0], value[1], value[2], value[3]];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttribI4i = function() {
+    this.caseName = 'vertex_attrib_4i';
+    this.name = 'VertexAttribI4i';
+    this.signed = true;
+    this.load = function(index, value) {
+        var v = new Int32Array(value);
+        gl.vertexAttribI4i(index, v[0], v[1], v[2], v[3]);
+        return [v[0], v[1], v[2], v[3]];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttribI4iv = function() {
+    this.caseName = 'vertex_attrib_4iv';
+    this.name = 'VertexAttribI4iv';
+    this.signed = true;
+    this.load = function(index, value) {
+        var v = new Int32Array(value);
+        gl.vertexAttribI4iv(index, v);
+        return [v[0], v[1], v[2], v[3]];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttribI4ui = function() {
+    this.caseName = 'vertex_attrib_4ui';
+    this.name = 'VertexAttribI4ui';
+    this.signed = false;
+    this.load = function(index, value) {
+        var v = new Uint32Array(value);
+        gl.vertexAttribI4ui(index, v[0], v[1], v[2], v[3]);
+        return [v[0], v[1], v[2], v[3]];
+    };
+};
+
+/**
+ * @constructor
+ */
+es3fDefaultVertexAttributeTests.LoaderVertexAttribI4uiv = function() {
+    this.caseName = 'vertex_attrib_4uiv';
+    this.name = 'VertexAttribI4uiv';
+    this.signed = false;
+    this.load = function(index, value) {
+        var v = new Uint32Array(value);
+        gl.vertexAttribI4uiv(index, v);
+        return [v[0], v[1], v[2], v[3]];
+    };
+};
+
+/** @const */ var RENDER_SIZE = 32;
+/** @const */ var s_valueRange = 10;
+/** @const */ var s_passThroughFragmentShaderSource = '#version 300 es\n' +
+                                                        'layout(location = 0) out mediump vec4 fragColor;\n' +
+                                                        'in mediump vec4 v_color;\n' +
+                                                        'void main (void)\n' +
+                                                        '{\n' +
+                                                        ' fragColor = v_color;\n' +
+                                                        '}\n';
```

Comment:
It's unfortunate that there's duplication like this shader in the dEQP tests. However, I don't think it would be a good idea to try to use WebGLTestUtils' shaders here, since that would introduce an undesirable dependency preventing easy tracking of the C++ code.